OpenMLDB/src/src/main/java/com/_4paradigm/benchmark/Util.java

```diff
@@ -31,49 +34,76 @@
+        Logger logger = LoggerFactory.getLogger(Util.class);
+
+        System.out.println(sql);
         java.sql.Statement state = executor.getStatement();
         try {
+            // Format the date and time using a formatter
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
+
             boolean ret = state.execute(sql);
             // check status
             ResultSet res = state.getResultSet();
             res.next();
-            System.out.println("[INFO LOG]:\n" + res.getString(1));
+            // print the job info
+            System.out.println("[INFO LOG]: "+LocalDateTime.now().format(formatter)+"\n" + res.getString(1));
             String jobID = extractJobID(res.getString(1));
-            System.out.println("[INFO LOG]:" + jobID);
-            boolean retOfShow = state.execute("SHOW JOB " + jobID);
+            boolean retOfShow = state.execute("SHOW JOB " + jobID + ";");
             ResultSet resOfShow = state.getResultSet();
             resOfShow.next();
+            
+            long startTime = System.currentTimeMillis();
+            long endTime = System.currentTimeMillis();
+
             while(!resOfShow.getString(3).toLowerCase().equals("finished") &&
                 !resOfShow.getString(3).toLowerCase().equals("lost") &&
                 !resOfShow.getString(3).toLowerCase().equals("killed") &&
                 !resOfShow.getString(3).toLowerCase().equals("failed")) {
-                System.out.println("[INFO LOG]:" + resOfShow.getString(6) + "///" + resOfShow.getString(3));
+
+                endTime = System.currentTimeMillis();
+                if ((endTime - startTime) / 1000 > 3600*3) {
+                    throw new TimeoutException("It appears that the loading process is taking too much time to complete. If you have a clear understanding of the issue, you may ignore this exception and remove the code that triggers it.");
+                }
+
+                // heartbeat
+                logger.debug("[HEART BEAT]: " + LocalDateTime.now().format(formatter) + " Job" + jobID + " " + resOfShow.getString(3));
+
                 resOfShow.close();
                 TimeUnit.SECONDS.sleep(30);
-                retOfShow = state.execute("SHOW JOB " + jobID);
+                retOfShow = state.execute("SHOW JOB " + jobID + ";");
                 resOfShow = state.getResultSet();
                 resOfShow.next();
             }
-            System.out.println("[INFO LOG]:" + resOfShow.getString(6) + "///" + resOfShow.getString(3));
-            resOfShow.close();
+            System.out.println("[INFO LOG]: " + LocalDateTime.now().format(formatter) + " Job" + jobID + " " + resOfShow.getString(3));
 
+            if(!resOfShow.getString(3).toLowerCase().equals("finished")) {
+                resOfShow.close();
+                throw new Exception("[ERROR]: loading failed with final state '"+ resOfShow.getString(3) +"', please check the job log");
```

Comment:
It’s better to make it a config, job Wait timeout, add a todo comment? Let’s add it later